src/object_store/aws.rs

```diff
@@ -537,4 +537,117 @@ mod tests {
     }
+
+    #[tokio::test]
+    async fn test_ctx_register_object_store() -> Result<()> {
+        let amazon_s3_file_system = Arc::new(
+            AmazonS3FileSystem::new(
+                Some(SharedCredentialsProvider::new(Credentials::new(
+                    ACCESS_KEY_ID,
+                    SECRET_ACCESS_KEY,
+                    None,
+                    None,
+                    PROVIDER_NAME,
+                ))),
+                None,
+                Some(Endpoint::immutable(Uri::from_static(MINIO_ENDPOINT))),
+                None,
+                None,
+                None,
+            )
+            .await,
+        );
+
+        let ctx = ExecutionContext::new();
+
+        ctx.register_object_store("s3", amazon_s3_file_system);
+
+        let (_, name) = ctx.object_store("s3").unwrap();
+        assert_eq!(name, "s3");
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[should_panic(expected = "NoSuchBucket")]
+    async fn test_read_nonexistent_bucket() {
+        let amazon_s3_file_system = AmazonS3FileSystem::new(
+            Some(SharedCredentialsProvider::new(Credentials::new(
+                ACCESS_KEY_ID,
+                SECRET_ACCESS_KEY,
+                None,
+                None,
+                PROVIDER_NAME,
+            ))),
+            None,
+            Some(Endpoint::immutable(Uri::from_static(MINIO_ENDPOINT))),
+            None,
+            None,
+            None,
+        )
+        .await;
+
+        let mut files = amazon_s3_file_system
+            .list_file("nonexistent_data")
+            .await
+            .unwrap();
+
+        while let Some(file) = files.next().await {
+            let sized_file = file.unwrap().sized_file;
+            let mut reader = amazon_s3_file_system
+                .file_reader(sized_file.clone())
+                .unwrap()
+                .sync_chunk_reader(0, sized_file.size as usize)
+                .unwrap();
+
+            let mut bytes = Vec::new();
+            let size = reader.read_to_end(&mut bytes).unwrap();
+
+            assert_eq!(size as u64, sized_file.size);
+        }
+    }
+
+    #[tokio::test]
+    #[should_panic(expected = "File is missing")]
+    async fn test_read_nonexistent_range() {
+        let start = 10;
+        let length = 128;
+
+        let amazon_s3_file_system = AmazonS3FileSystem::new(
+            Some(SharedCredentialsProvider::new(Credentials::new(
+                ACCESS_KEY_ID,
+                SECRET_ACCESS_KEY,
+                None,
+                None,
+                PROVIDER_NAME,
+            ))),
+            None,
+            Some(Endpoint::immutable(Uri::from_static(MINIO_ENDPOINT))),
+            None,
+            None,
+            None,
+        )
+        .await;
+        let mut files = amazon_s3_file_system
+            .list_file("data/nonexistent_file.txt")
+            .await
+            .unwrap();
+
+        if let Some(file) = files.next().await {
```

Comment:
Could you just do something like this below (untested) and remove the panic (and all this other non-used code)?

```
assert_eq!(files.next().await, None)
```

Reply:
Thats much nicer, will do